spinta/datasets/backends/sql/commands/read.py

```diff
@@ -105,10 +102,18 @@ def getall(
                     if isinstance(sel.prop.dtype, PrimaryKey):
                         val = keymap.encode(sel.prop.model.model_type(), val)
                     elif isinstance(sel.prop.dtype, Ref):
-                        val = keymap.encode(sel.prop.dtype.model.model_type(), val)
+                        if sel.prop.dtype.refprops == sel.prop.dtype.model.external.pkeys or len(
+                            sel.prop.dtype.model.external.pkeys) == 1:
```

Comment:
Why do you need this second condition? I think this is the reason why I get incorrect key on my testing notes.

Reply:
It because of this test
tests/datasets/test_sql.py::test_composite_ref_keys

it seems it should be 
val = keymap.encode(sel.prop.model.model_type(), val)

but condition is not equal
sel.prop.dtype.refprops == sel.prop.dtype.model.external.pkeys

so i added additional condition
sel.prop.dtype.refprops == sel.prop.dtype.model.external.pkeys or len(
                            sel.prop.dtype.refprops) > len(sel.prop.dtype.model.external.pkeys):